pkg/osinfo/container_scanner.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package osinfo
+
+import (
+	"bufio"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type ContainerScanner struct{}
+
+// ReadOSPackages reads a bunch of layers and extracts the os package
+// information from them, it returns the OS package and the layer where
+// they are defined.
+func (ct *ContainerScanner) ReadOSPackages(layers []string) (
+	layerNum int, packages *[]PackageDBEntry, err error,
+) {
+	loss := LayerScanner{}
+
+	// First, let's try to determine which OS the container is based on
+	osKind := ""
+	for _, lp := range layers {
+		osKind, err = loss.OSType(lp)
+		if err != nil {
+			return 0, nil, errors.Wrap(err, "reading os type from layer")
+		}
+		if osKind != "" {
+			break
+		}
+	}
+
+	if osKind == OSDebian {
+		return ct.ReadDebianPackages(layers)
+	}
+	return 0, nil, nil
```

Comment:
Do we want to error in case of an unsupported OS?

Reply:
Not really. An unsupported OS should not mean that we cannot produce an SBOM. I think the list pointer makes sense as in "no, error but nothing found". Updated the documentation a bit more to state that.